rustworkx/rustworkx.pyi

```diff
@@ -1079,7 +1080,9 @@ class _RustworkxCustomVecIter(Generic[_T_co], Sequence[_T_co], ABC):
     def __len__(self) -> int: ...
     def __ne__(self, other: object) -> bool: ...
     def __setstate__(self, state: Sequence[_T_co]) -> None: ...
-    def __array__(self, dtype: np.dtype | None = ..., copy: bool | None = ...) -> np.ndarray: ...
+    def __array__(
+        self, dtype: np.dtype[Any] | None = ..., copy: bool | None = ...
+    ) -> npt.NDArray[Any]: ...
```

Comment:
Do we want to use a generic/typevar here? Since the `Any` for the dtype argument needs to match the dtype inside the returned array?

Reply:
For the record we ignore the `dtype` at runtime. However, I cannot annotate the ndarray with `_T` as that would be incorrect. There is a mapping from `_T` to numpy types that we’d need to manually add and that is a lot of work.

Also, this method exists to support `np.array` calls no one should call it directly. I think NumPy type stubs will mask this call anyway